diff --git a/BUILD b/BUILD
new file mode 100644
index 0000000..b430515
--- /dev/null
+++ b/BUILD
@@ -0,0 +1,56 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+cc_library(
+    name = "leveldb",
+    srcs = glob(
+        [
+            "db/**/*.cc",
+            "db/**/*.h",
+            "helpers/**/*.cc",
+            "helpers/**/*.h",
+            "port/**/*.cc",
+            "port/**/*.h",
+            "table/**/*.cc",
+            "table/**/*.h",
+            "util/**/*.cc",
+            "util/**/*.h",
+        ],
+        exclude = [
+            "**/*_test.cc",
+            "**/testutil.*",
+            "**/*_bench.cc",
+            "**/*_windows*",
+            "db/leveldbutil.cc",
+        ],
+    ),
+    copts = ['-DLEVELDB_PLATFORM_POSIX=1'],
+    hdrs = glob(
+        ["include/**/*.h"],
+        exclude = ["doc/**"],
+    ) + glob(
+        ["port/*.h"]
+    ),
+    includes = [
+        "include",
+    ],
+    visibility = ["//visibility:public"],
+    deps = [
+        "@crc32c//:crc32c",
+        "@snappy//:snappy",
+    ],
+)
diff --git a/port/port_stdcxx.h b/port/port_stdcxx.h
index 2bda48d..cefacde 100644
--- a/port/port_stdcxx.h
+++ b/port/port_stdcxx.h
@@ -26,7 +26,7 @@
 #include <crc32c/crc32c.h>
 #endif  // HAVE_CRC32C
 #if HAVE_SNAPPY
-#include <snappy.h>
+#include "snappy.h"
 #endif  // HAVE_SNAPPY
 
 #include <cassert>
