diff --git a/BUILD b/BUILD
new file mode 100644
index 00000000..40efde99
--- /dev/null
+++ b/BUILD
@@ -0,0 +1,48 @@
+package(
+  default_visibility = ['//:__pkg__'],
+)
+
+load('@rules_foreign_cc//foreign_cc:defs.bzl', 'cmake')
+
+filegroup(
+  name = 'all_srcs',
+  srcs = glob(['**'], exclude = ['bazel-*/**']),
+)
+
+SYMLINK2REAL_SCRIPT = """bash -c '
+set -ex
+shopt -s globstar
+for file in $INSTALLDIR/** ; do
+  if [ -L $file ]; then
+    target=`readlink -f $file`
+    echo $file $target
+    rm $file
+    cp $target $file
+  fi
+done
+'"""
+
+cmake(
+  name = 'event',
+  lib_source = ':all_srcs',
+  cache_entries = {
+    'EVENT__DISABLE_BENCHMARK': 'ON',
+    'EVENT__DISABLE_TESTS': 'ON',
+    'EVENT__DISABLE_SAMPLES': 'ON',
+    'EVENT__LIBRARY_TYPE': 'STATIC',
+    'OPENSSL_ROOT_DIR': '$$EXT_BUILD_DEPS$$/openssl',
+  },
+  postfix_script = SYMLINK2REAL_SCRIPT,
+  out_static_libs = [
+    'libevent.a',
+    'libevent_core.a',
+    'libevent_extra.a',
+    'libevent_openssl.a',
+    'libevent_pthreads.a',
+  ],
+  visibility = ['//visibility:public'],
+  deps = [
+    '@openssl//:crypto',
+    '@openssl//:ssl',
+  ],
+)
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 676727f1..ef5645d4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -955,7 +955,7 @@ add_event_library(event_extra
 if (NOT EVENT__DISABLE_OPENSSL)
     add_event_library(event_openssl
         INNER_LIBRARIES event_core
-        OUTER_INCLUDES ${OPENSSL_INCLUDE_DIR}
+        OUTER_INCLUDES $<BUILD_INTERFACE:${OPENSSL_INCLUDE_DIR}>
         LIBRARIES ${OPENSSL_LIBRARIES}
         SOURCES ${SRC_OPENSSL})
 endif()
